bigtable-hbase-parent/bigtable-hbase-mapreduce/pom.xml

```diff
@@ -33,54 +33,125 @@ limitations under the License.
     </description>
 
     <properties>
-        <hbase.version>1.1.2</hbase.version>
+        <hbase.version>1.2.4</hbase.version>
+        <hadoop.version>2.7.3</hadoop.version>
     </properties>
 
     <dependencies>
         <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-tcnative-boringssl-static</artifactId>
+            <scope>compile</scope>
         </dependency>
+
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-common</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.hbase</groupId>
-            <artifactId>hbase-annotations</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.hbase</groupId>
-            <artifactId>hbase-common</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.hbase</groupId>
-            <artifactId>hbase-client</artifactId>
+            <artifactId>hadoop-client</artifactId>
+            <version>${hadoop.version}</version>
+            <scope>provided</scope>
         </dependency>
+
         <dependency>
-            <groupId>com.google.cloud.bigtable</groupId>
-            <artifactId>bigtable-hbase-1.1</artifactId>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>bigtable-hbase-1.2</artifactId>
             <version>${project.version}</version>
+
             <exclusions>
                 <exclusion>
-                    <groupId>${project.groupId}</groupId>
-                    <artifactId>bigtable-hbase-shaded</artifactId>
+                    <groupId>io.netty</groupId>
```

Comment:
I should've added a comment explaining this. Its a not a question if these deps are used, but that shading them would create a conflict with hadoop's classpath. Since hadoop (< 3.0) doesn't isolate its classpath, mapreduce jobs live in hadoop's classpath. I'm only trying to de-duplicate the deps on the classpath.

To make this situation somewhat manageable, hadoop publishes hadoop-client pom.xml which lists all of the deps as provided. Unfortunately maven doesn't have a way to force conflict resolution in favor of the provided scope and mixes the deps between hadoop-client's provided deps and hbase-client's compile deps.

The easiest way I found to discover the overlap is to list hadoop-client first and exclude all hadoop-client deps that are marked as compile (since hadoop-client is listed first, it will keep its deps and if maven picks hbase-client's version it will inherit the compile scope).  Using this approach, we will be able upgrade the hbase dep in the future. Switching it to a shade inclusion model, while less verbose, will make it harder to upgrade hbase.